apps/root/src/common/components/token-selector/index.tsx

```diff
@@ -0,0 +1,107 @@
+import styled from 'styled-components';
+import React from 'react';
+import find from 'lodash/find';
+import { Chip, ContainerBox, Select, Typography, colors } from 'ui-library';
+import TokenIcon from '@common/components/token-icon';
+import useSelectedNetwork from '@hooks/useSelectedNetwork';
+import { formatCurrencyAmount } from '@common/utils/currency';
+import useActiveWallet from '@hooks/useActiveWallet';
+import { Token } from '@types';
+import { TokenBalance, useWalletBalances } from '@state/balances/hooks';
+import useTokenList from '@hooks/useTokenList';
+import { formatUnits } from 'viem';
+import { useThemeMode } from '@state/config/hooks';
+import { defineMessage, useIntl } from 'react-intl';
+
+interface TokenSelectorProps {
+  handleChange: (token: Token) => void;
+  selectedToken?: Nullable<Token>;
+}
+
+const StyledNetworkContainer = styled.div`
+  display: flex;
+  gap: 8px;
+  flex-direction: column;
+`;
+
+const StyledNetworkButtonsContainer = styled.div`
+  display: flex;
+  gap: 10px;
+  align-items: center;
+  flex-wrap: wrap;
+`;
+
+type OptionWithKeyAndToken = TokenBalance & { key: string; token: Token };
+
+const TokenItem = ({ item: { token, balance, balanceUsd, key } }: { item: OptionWithKeyAndToken }) => {
+  const mode = useThemeMode();
+
+  return (
+    <ContainerBox alignItems="center" key={key} flex={1} gap={3}>
+      <TokenIcon size={7} token={token} />
+      <ContainerBox flexDirection="column" flex="1">
+        <Typography variant="body" fontWeight={600} color={colors[mode].typography.typo2}>
+          {token.name}
+        </Typography>
+        <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+          {formatCurrencyAmount(balance, token)}
+          {` `}
+          {token.symbol}
+        </Typography>
+      </ContainerBox>
+      {!!balanceUsd && (
+        <Chip size="small" label={`$${parseFloat(formatUnits(balanceUsd, token.decimals + 18)).toFixed(2)}`} />
+      )}
+    </ContainerBox>
+  );
+};
+
+const searchFunction = ({ token }: OptionWithKeyAndToken, search: string) =>
+  token.name.toLowerCase().includes(search.toLowerCase()) ||
+  token.symbol.toLowerCase().includes(search.toLowerCase()) ||
+  token.address.toLowerCase().includes(search.toLowerCase());
+
+const TokenSelector = ({ handleChange, selectedToken }: TokenSelectorProps) => {
+  const activeWallet = useActiveWallet();
+  const intl = useIntl();
+  const selectedNetwork = useSelectedNetwork();
+  const { balances } = useWalletBalances(activeWallet?.address, selectedNetwork.chainId);
+
+  const tokens = useTokenList({ allowAllTokens: true, filterChainId: true, filter: false });
+
+  const availableTokens = React.useMemo(
+    () =>
+      Object.keys(balances).map<OptionWithKeyAndToken>((tokenAddress) => ({
```

Comment:
I believe this is working as expected, but when having no balances at all in the selectedNetwork, the the token selector is empty, and I think it can be confusing

![Screenshot 2024-02-14 at 10 07 51](https://github.com/Mean-Finance/dca-fe/assets/144699868/a3de2741-795e-429d-8b83-e6e6242552bc)

The input doesn't have a label or placeholder, so in this case, it's not clear what is this input for. Also, when opening the list, it's just empty.
I'm thinking we could add a default option saying 'You have no balance in the selected network" + Disable input